File: patterns/c++/LevelOrderTraversal.cpp
#include <iostream>
#include <queue>
using namespace std;

// Definition for a binary tree node.
struct TreeNode {
    int val;
    TreeNode* left;
    TreeNode* right;

    TreeNode(int x) : val(x), left(nullptr), right(nullptr) {}
};

class LevelOrderTraversal {
public:
    void levelOrder(TreeNode* root) {
        if (root == nullptr) return;

        queue<TreeNode*> q;
        q.push(root);

        while (!q.empty()) {
            TreeNode* node = q.front();
            q.pop();
            cout << node->val << " "; // Process the node by printing its value
            
            // Add the left and right children to the queue, if they exist
            if (node->left != nullptr) q.push(node->left);
            if (node->right != nullptr) q.push(node->right);
        }
    }
};

File: patterns/python/level_order_traversal.py
from collections import deque

# Definition for a binary tree node.
class TreeNode:
    def __init__(self, x):
        self.val = x
        self.left = None
        self.right = None

class LevelOrderTraversal:
    def level_order(self, root):
        if root is None:
            return

        queue = deque([root])

        while queue:
            node = queue.popleft()
            print(node.val, end=" ")  # Process the node by printing its value
            
            # Add the left and right children to the queue, if they exist
            if node.left:
                queue.append(node.left)
            if node.right:
                queue.append(node.right)